### PR TITLE
Feat/bulk frontend

### PR DESCRIPTION
**Pull request by gweiying**
## Problem

_What problem are you trying to solve? What issue does this close?_
https://www.notion.so/opengov/c622b311c6ae4ef0b416ce47decd0874?v=8df8c11db2944cd08ffc774e3fbae392&p=577c91546dcc4f349e1e6927d5476fdc&pm=s

## Solution

_How did you solve the problem?_
Created bulk link upload form with downloadable csv.

Other changes:
- Create `CreateType` enum, with enum values `LINK, FILE, BULK` to store current displayed creation modal
- Create `CreateTypeButton` to abstract over the different types of creation modal

## Before & After Screenshots

**AFTER**:

Modal with `From a .csv` option
Screenshot 2022-09-01 at 19 46 11

Modal with `From a .csv` option selected
Screenshot 2022-09-01 at 19 46 26

File selection - note that only csv files can be selected
Screenshot 2022-09-01 at 19 49 54

Template csv fields
Screenshot 2022-09-01 at 19 50 17

### CODE REVIEW

**Review by halfwhole — COMMENTED** [review]
basically lgtm, have only a few small things to add!

I was a bit confused though about the terminology for "from" and "to" on the create type buttons - shouldn't the options for creating a regular URL/file be "to" a URL/file, since our shortlinks lead us to an external URL/file, reserving the word "from" for creating shortlinks from .csv files? (might be a design issue)

also, will we be adding the "New!" chip to the "From a .csv" tab?

File: src/client/user/components/CreateUrlModal/components/CreateTypeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React, { ComponentType, FunctionComponent } from 'react'
+import { Button, Typography, useTheme } from '@material-ui/core'
+
+type IconProps = any
+type CreateTypeButtonProps = {
+  InputProps: any
+  Icon: ComponentType<IconProps>
+  isEnabled: boolean
+  onChange: () => void
+  childen: string
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/client/user/components/CreateUrlModal/components/CreateTypeButton.tsx]: [6f3df83](https://github.com/opengovsg/GoGovSG/pull/1948/commits/6f3df8337b4eda00e308d518ffec3336274c1c3b)

File: src/client/user/components/CreateUrlModal/CreateLinkForm.tsx [hunk 2]
```diff
@@ -67,32 +77,56 @@ const CreateLinkForm: FunctionComponent<CreateLinkFormProps> = ({
...
   const history = useHistory()
 
-  const [isFile, setIsFile] = useState(false)
+  const [createType, setCreateType] = useState<CreateType>(CreateType.LINK)
   const [file, setFile] = useState<File | null>(null)
 
   const classes = useCreateLinkFormStyles({
     textFieldHeight: TEXT_FIELD_HEIGHT,
-    isFile,
+    isFile: createType === CreateType.FILE,
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/client/user/components/CreateUrlModal/CreateLinkForm.tsx]: good catch, [a91f211](https://github.com/opengovsg/GoGovSG/pull/1948/commits/a91f211731a128d1882c8464511c46aa525e80b8)

File: src/client/user/components/CreateUrlModal/CreateLinkForm.tsx [hunk 3]
```diff
@@ -236,77 +257,164 @@ const CreateLinkForm: FunctionComponent<CreateLinkFormProps> = ({
...
+                  how:
+                  <br />
+                  {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+                  1.{' '}
+                  <Button
+                    onClick={downloadSampleBulkCsv}
+                    className={classes.bulkSampleDownloadText}
+                  >
+                    Download template .csv file
+                  </Button>
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/client/user/components/CreateUrlModal/CreateLinkForm.tsx:348]: [5d088fb](https://github.com/opengovsg/GoGovSG/pull/1948/commits/5d088fb9a062925d0ec5da85ee8bfbd67b378575) - gave it my best shot, let me know if you have better css ideas ._.

**Comment by gweiying** [comment]
> I was a bit confused though about the terminology for "from" and "to" on the create type buttons - shouldn't the options for creating a regular URL/file be "to" a URL/file, since our shortlinks lead us to an external URL/file, reserving the word "from" for creating shortlinks from .csv files? (might be a design issue)

Good idea, implemented in [5d088fb](https://github.com/opengovsg/GoGovSG/pull/1948/commits/5d088fb9a062925d0ec5da85ee8bfbd67b378575)

> also, will we be adding the "New!" chip to the "From a .csv" tab?
Checked with Feli, she's ok w the current design!

**Review by halfwhole — APPROVED** [review]
looks great!

**Pull request merged**